main_parser.go

```diff
@@ -0,0 +1,460 @@
+package SQL_parser
```

Comment:
To start with this file needs to go into its own package i.e perhaps in `pkg/select` and expose APIs from there. 

Start following the style recommended here in all the code, including code comments etc https://github.com/golang/go/wiki/CodeReviewComments